tests/integration/base_deploy_test.go

```diff
@@ -73,14 +76,34 @@ func checkIfRookClusterIsHealthy(s suite.Suite, testClient *clients.TestClient,
 	require.Nil(s.T(), err)
 }
 
-func gatherAllRookLogs(k8sh *utils.K8sHelper, s suite.Suite, hostType string, opNamespace string, clusterNamespace string) {
-	logger.Infof("Gathering all logs from Rook Cluster %s", clusterNamespace)
-	k8sh.GetRookLogs("rook-operator", hostType, opNamespace, s.T().Name())
-	k8sh.GetRookLogs("rook-agent", hostType, opNamespace, s.T().Name())
-	k8sh.GetRookLogs("rook-api", hostType, clusterNamespace, s.T().Name())
-	k8sh.GetRookLogs("rook-ceph-mgr", hostType, clusterNamespace, s.T().Name())
-	k8sh.GetRookLogs("rook-ceph-mon", hostType, clusterNamespace, s.T().Name())
-	k8sh.GetRookLogs("rook-ceph-osd", hostType, clusterNamespace, s.T().Name())
-	k8sh.GetRookLogs("rook-ceph-rgw", hostType, clusterNamespace, s.T().Name())
-	k8sh.GetRookLogs("rook-ceph-mds", hostType, clusterNamespace, s.T().Name())
+func HandlePanics(r interface{}, o contracts.TestOperator, t func() *testing.T) {
+	if r != nil {
+		logger.Infof("unexpected panic occurred during test %s, --> %v", t().Name(), r)
+		t().Fail()
+		o.TearDown()
+		t().FailNow()
+	}
+
+}
+
+//BaseTestOperations struct for handling panic and test suite tear down
+type BaseTestOperations struct {
+	installer     *installer.InstallHelper
+	T             func() *testing.T
+	namespace     string
+	helmInstalled bool
+}
+
+//GetBaseTestOperations creates new instance of BaseTestOperations struct
+func GetBaseTestOperations(i *installer.InstallHelper, t func() *testing.T, namespace string, helmInstalled bool) BaseTestOperations {
```

Comment:
nit: it may be more idiomatic golang to call this `NewBaseTestOperations`

Reply:
haha. I was calling it `NewBaseTestOperations` , but some how i convinced myself to change it.